precompiles/avs/query.go

```diff
@@ -60,3 +62,49 @@ func (p Precompile) GetOptedInOperatorAccAddrs(
 	}
 	return method.Outputs.Pack(list)
 }
+
+// GetAVSUSDValue is a function to retrieve the USD share of specified Avs,
+func (p Precompile) GetAVSUSDValue(
+	ctx sdk.Context,
+	_ *vm.Contract,
+	method *abi.Method,
+	args []interface{},
+) ([]byte, error) {
+	if len(args) != len(p.ABI.Methods[MethodGetAVSUSDValue].Inputs) {
+		return nil, fmt.Errorf(cmn.ErrInvalidNumberOfArgs, 1, len(args))
+	}
+	addr, ok := args[0].(common.Address)
+	if !ok {
+		return nil, fmt.Errorf(exocmn.ErrContractInputParaOrType, 0, "string", addr)
+	}
+	amount, err := p.avsKeeper.GetOperatorKeeper().GetAVSUSDValue(ctx, addr.String())
+	if err != nil {
+		return nil, err
+	}
+	return method.Outputs.Pack(amount.BigInt())
+}
+
+// GetOperatorOptedUSDValue is a function to retrieve the USD share of specified operator and Avs,
+func (p Precompile) GetOperatorOptedUSDValue(
+	ctx sdk.Context,
+	_ *vm.Contract,
+	method *abi.Method,
+	args []interface{},
+) ([]byte, error) {
+	if len(args) != len(p.ABI.Methods[MethodGetOperatorOptedUSDValue].Inputs) {
+		return nil, fmt.Errorf(cmn.ErrInvalidNumberOfArgs, 1, len(args))
+	}
+	avsAddr, ok := args[0].(common.Address)
+	if !ok {
+		return nil, fmt.Errorf(exocmn.ErrContractInputParaOrType, 0, "string", avsAddr)
+	}
+	operatorAddr, ok := args[1].(string)
+	if !ok {
+		return nil, fmt.Errorf(exocmn.ErrContractInputParaOrType, 1, "string", operatorAddr)
+	}
+	amount, err := p.avsKeeper.GetOperatorKeeper().GetOperatorOptedUSDValue(ctx, avsAddr.String(), operatorAddr)
+	if err != nil {
+		return nil, err
+	}
+	return method.Outputs.Pack(amount.ActiveUSDValue.BigInt())
```

Comment:
Has this been tested to be compatible with Solidity? For example does a valid of 1 become the uint256 1 or 1000000000... (or maybe in binary) ?

Reply:
testing has been added, s.Require().Equal(expectedUSDvalue.BigInt(), out),meets expectations